pkg/certificates/reconciler/certificates.go

```diff
@@ -95,16 +105,29 @@ func (r *reconciler) ReconcileKind(ctx context.Context, secret *corev1.Secret) p
 	cert, _, err := parseAndValidateSecret(secret, true)
 	if err != nil {
 		r.logger.Infof("Secret invalid: %v", err)
-		sans := []string{certificates.DataPlaneNamePrefix + secret.Namespace, certificates.LegacyFakeDnsName}
 		// Check the secret to reconcile type
-		var keyPair *certificates.KeyPair
-		if secret.Labels[r.secretTypeLabelName] == dataPlaneSecretType {
-			keyPair, err = certificates.CreateCert(ctx, caPk, caCert, expirationInterval, sans...)
-		} else if secret.Labels[r.secretTypeLabelName] == controlPlaneSecretType {
-			keyPair, err = certificates.CreateCert(ctx, caPk, caCert, expirationInterval, sans...)
-		} else {
+
+		pipelineId, ok := secret.Labels[r.secretPipelineId]
```

Comment:
Is it okay for the label to be the empty string?

If not, you can check `!pipelineId`